lib/bademagic_module/utils/converters.dart

```diff
@@ -43,6 +43,20 @@ class Converters {
     return hexStrings;
   }
 
+  void badgeAnimation(String message) async {
+    if (message == "") {
+      //geerate a 2d list with all values as 0
+      List<List<int>> image =
+          List.generate(11, (i) => List.generate(44, (j) => 0));
+      badgeList.setNewGrid(image);
+    } else {
+      List<String> hexStrings = await messageTohex(message);
```

Comment:
We should avoid these 3 operations and decide a scheme to store messages to one format which can be used across the app. List<List<bool>>  would be a good candidate. It does not make sense to have these translations for every step when we could built our logic just on one kind of data

Reply:
But in order to get the bitmap (List<List<int>>) we would first need to have the hexstring and then convert it to bitmap.
Can you clarify what you mean by avoiding the operations.

Reply:
Hex String -> Byte Array, Byte Array -> Binary Array, Binary Array -> Bool Array. Why not Hex String -> Bool Array

Reply:
I have not seen this conversion so far si it possible?

Reply:
Give me some time I will look into it and if I can make it then it will be very efficient.

Reply:
So we're trying to minimize the number of conversions. HexString to something you can work off of everywhere. Maybe 2 conversions is fine but not more

Reply:
So HexString -> ByteArray is fine. Then maybe ByteArray -> Bool Array when needed.

Reply:
I don't see a point in holding up 4 bytes of memory whereas you can get away with just a bit

Reply:
Correction, flutter uses 8 bytes int so more memory usage for no reason